src/l500/l500-device.cpp

```diff
@@ -657,45 +665,36 @@ namespace librealsense
             if (res.size() < expected_size)
             {
                 throw std::runtime_error(
-                    to_string() << "TEMPERATURES_GET - Invalid result size!, expected: "
+                    to_string() << "TEMPERATURES_GET - Invalid result size! expected: "
                     << expected_size << " bytes, "
                     "got: " << res.size() << " bytes");
             }
 
-            auto t = reinterpret_cast<temperatures*>((void*)res.data());
-            rv.LDD_temperature = t->LDD_temperature;
-            rv.MC_temperature = t->MC_temperature;
-            rv.MA_temperature = t->MA_temperature;
-            rv.APD_temperature = t->APD_temperature;
-            rv.HUM_temperature = t->HUM_temperature;
-            rv.AlgoTermalLddAvg_temperature = t->AlgoTermalLddAvg_temperature;
-
             if (fw_version_support_nest)
-            {
-                extended_temperatures *t_ex = reinterpret_cast<extended_temperatures*>((void*)res.data());
-                rv.nest_avg = t_ex->nest_avg;
-            }
-
+                rv = *reinterpret_cast<extended_temperatures const *>(res.data());
+            else
+                *reinterpret_cast<temperatures *>(&rv) = *reinterpret_cast<temperatures const *>(res.data());
         }
         return rv;
     }
 
-    void l500_device::start_temperatures_fetcher()
+    void l500_device::start_temperatures_reader()
     {
         LOG_DEBUG("Starting temperature fetcher thread");
-        _temperature_fetcher = std::thread( [&]() {
+        _temperature_reader = std::thread( [&]() {
             try
             {
-                auto fw_version_support_nest = ( _fw_version >= firmware_version( "1.5.0.0" ) ) ? true : false;
+                auto fw_version_support_nest = _fw_version >= firmware_version( "1.5.0.0" );
+
                 auto expected_size = fw_version_support_nest ? sizeof( extended_temperatures )
                                                              : sizeof( temperatures );
 
                 utilities::time::periodic_timer pt(std::chrono::seconds(1));
                 pt.set_expired(); // Force condition true on start
 
-                _temperature_fetcher_on = true;
+                _keep_reading_temperature = true;
```

Comment:
Should we move this out of the thread (right before line 684) in case we try to cancel it before we get to the while in the next line?

Reply:
👍